jwst/photom/photom.py

```diff
@@ -34,52 +36,53 @@ def __init__(self, model):
             input Data Model object
 
         """
-        self.input = model
+        # Create a copy of the input model
+        self.input = model.copy()
 
         self.instrument = model.meta.instrument.name.upper()
         self.detector = model.meta.instrument.detector.upper()
         self.exptype = model.meta.exposure.type.upper()
         self.filter = None
-        if model.meta.instrument.filter is not None:
+        if model.meta.instrument.filter:
             self.filter = model.meta.instrument.filter.upper()
         self.pupil = None
-        if model.meta.instrument.pupil is not None:
+        if model.meta.instrument.pupil:
             self.pupil = model.meta.instrument.pupil.upper()
         self.grating = None
-        if model.meta.instrument.grating is not None:
+        if model.meta.instrument.grating:
             self.grating = model.meta.instrument.grating.upper()
         self.band = None
-        if model.meta.instrument.band is not None:
+        if model.meta.instrument.band:
             self.band = model.meta.instrument.band.upper()
         self.slitnum = -1
 
+        # Let the user know what we're working with
         log.info('Using instrument: %s', self.instrument)
         log.info(' detector: %s', self.detector)
         log.info(' exp_type: %s', self.exptype)
-        if self.filter is not None:
+        if self.filter:
             log.info(' filter: %s', self.filter)
-        if self.pupil is not None:
+        if self.pupil:
             log.info(' pupil: %s', self.pupil)
-        if self.grating is not None:
+        if self.grating:
             log.info(' grating: %s', self.grating)
-        if self.band is not None:
+        if self.band:
             log.info(' band: %s', self.band)
 
     def calc_nirspec(self, ftab, area_fname):
         """
         Extended Summary
         -------------
-        For the NIRSPEC instrument, extract PHOTMJSR from the input model to
-        write to the output model. Matching is based on FILTER and GRATING.
-
+        For the NIRSPEC instrument, reference file matching is based on
+        FILTER and GRATING, as well as SLIT name for the fixed-slits mode.
         The routine will find the corresponding information in the reference
-        file and write the conversion factor to the output model.  The routine
-        will also check to see if there is a wavelength-dependent response
-        table; if there is it will be written to the output model.
+        file, apply it to the data, and write the scalar conversion
+        factor to the output model. All NIRSpec modes use wavelength-dependent
+        flux calibration factors.
 
         Parameters
         ----------
-        ftab: fits HDUList
+        ftab: FITS HDUList
```

Comment:
`~astrpy.io.fits.HDUList`

Reply:
and needs space before the column

Reply:
In the process of updating these I noticed that these actually aren't an HDUList at all, but rather a jwst DataModel. So I've declared them all that way (and cleaned up the spacing of colons).